subprojects/dependency-management/src/main/java/org/gradle/api/internal/artifacts/ivyservice/moduleconverter/dependencies/DefaultLocalConfigurationMetadataBuilder.java

```diff
@@ -205,11 +206,14 @@ private DependencyState doGetDefinedState(ConfigurationInternal configuration, C
         ImmutableSet.Builder<LocalFileDependencyMetadata> fileBuilder = ImmutableSet.builder();
         ImmutableList.Builder<ExcludeMetadata> excludeBuilder = ImmutableList.builder();
 
+        // Configurations that are not declarable should not have dependencies or constraints present,
+        // but we need to allow dependencies to be checked to avoid emitting many warnings when the
```

Comment:
Does this mean kotlin is able to add dependencies to non-declarable configurations somehow? Why can't we add an if check and assert here? Can you provide some context here?

Reply:
This appears to be due to some smoke tests failing when running with test distribution due to a lack of an `ANDROID_SDK_ROOT` environment variable.  I'm going to reach out to dev prod about this.  Testing on CI with this removed now, I think I might have just been confused by smoke test failure with TD on vs. TD off.

Reply:
While this is a problem, there are other problems where the Kotlin plugin adds dependencies to configurations that aren't declarable.  See the failures here: https://builds.gradle.org/buildConfiguration/Gradle_Master_Check_Stage_PullRequestFeedback_Trigger/64527094?hideProblemsFromDependencies=false&hideTestsFromDependencies=false&expandBuildChangesSection=true&expandBuildTestsSection=true
This must remain.  I'll add a note about how to find these in failing tests.